src/c1_foeppl_von_karman/foeppl_von_karman_curvable_bell_elements.h

```diff
@@ -46,672 +46,1160 @@
+          {
+            double coeff = hess_of_transform[0](alpha, beta);
+            if (fabs(coeff) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_dwd1_index
+			<< std::endl;
+              left_data_used[left_dwd1_index] = true;
+              continue;
+            }
+          }
+          if (!left_dwd2_pinned && !left_data_used[left_dwd2_index])
+          {
+            double coeff = hess_of_transform[1](alpha, beta);
+            if (fabs(coeff) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_dwd2_index
+			<< std::endl;
+              left_data_used[left_dwd2_index] = true;
+              continue;
+            }
+          }
+          if (!left_dwd1d1_pinned && !left_data_used[left_dwd1d1_index])
+          {
+            double coef =
+              jac_of_transform(0, alpha) * jac_of_transform(0, beta);
+            if (fabs(coef) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_dwd1d1_index
+			<< std::endl;
+              left_data_used[left_dwd1d1_index] = true;
+              continue;
+            }
+          }
+          if (!left_dwd1d2_pinned && !left_data_used[left_dwd1d2_index])
+          {
+            double coef =
+              jac_of_transform(0, alpha) * jac_of_transform(1, beta) +
+              jac_of_transform(1, alpha) * jac_of_transform(0, beta);
+            if (fabs(coef) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_dwd1d2_index
+			<< std::endl;
+              left_data_used[left_dwd1d2_index] = true;
+              continue;
+            }
+          }
+          if (!left_dwd2d2_pinned && !left_data_used[left_dwd2d2_index])
+          {
+            double coef =
+              jac_of_transform(1, alpha) * jac_of_transform(1, beta);
+            if (fabs(coef) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_dwd2d2_index
+			<< std::endl;
+              left_data_used[left_dwd2d2_index] = true;
+              continue;
+            }
+          }
+          // -------------------------------------------------------------------
+          // If we made it here, it is because all dofs in the constraint are
+          // pinned so we need to check the constraint is satisfied manually and
+          // then remove it by pinning the corresponding lagrange multiplier
 
-    //----------------------------------------------------------------------
-    // Class construction
+          // hierher Aidan: should this remain alive?
```

Comment:
Same as previous.

Reply:
Aidan: please tidy up brutally; I'll keep this comment in there to remind you but will mark this as resolved